sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -180,6 +180,15 @@ class SparkSession private(
    */
   @transient lazy val conf: RuntimeConfig = new RuntimeConfig(sessionState.conf)
 
+  /**
+   * Same as `spark.default.parallelism`, can be isolated across sessions.
+   *
+   * @since 3.1.0
+   */
+  def defaultParallelism: Int = {
+    sessionState.conf.defaultParallelism.getOrElse(sparkContext.defaultParallelism)
```

Comment:
so we add a config, whose only usage is to let users get the config value?

Reply:
As I said above. If add this config, I will move the exists `defaultParallelism` which in sql module follow up. e.g. [FilePartition.maxSplitBytes()](https://github.com/apache/spark/blob/82ff29be7afa2ff6350310ab9bdf6b474398fdc1/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FilePartition.scala#L91)

Reply:
just do this in this pr ?

Reply:
please do, otherwise it's a useless config